clang/lib/StaticAnalyzer/Checkers/StreamChecker.cpp

```diff
@@ -98,6 +107,19 @@ struct StreamState {
     OpenFailed /// The last open operation has failed.
   } State;
 
+  StringRef getKindStr() const {
+    switch (State) {
+    case Opened:
+      return "Opened";
+    case Closed:
+      return "Closed";
+    case OpenFailed:
+      return "OpenFailed";
+    default:
+      llvm_unreachable("Unknown StreamState!");
```

Comment:
Having a default case is an anti-pattern that suppresses incomplete switch warnings if the enumeration ever changes.
I'd suggest dropping this switch-case.

```suggestion
```